packages/wonder-blocks-dropdown/components/dropdown-core.js

```diff
@@ -61,78 +77,171 @@ export default class DropdownCore extends React.Component<DropdownCoreProps> {
+        // attempting to handle this event (which would close the menu). The
+        // menu callback stops propagation of the event.
+        document.addEventListener("click", this._handleInteract);
+        document.addEventListener("touch", this._handleInteract);
+        // This is still keyup because we are listening for the escape key,
+        // which invokes events keydown and keyup and NOT keypress.
+        document.addEventListener("keyup", this._handleKeyup);
+    }
+
+    removeEventListeners() {
+        document.removeEventListener("click", this._handleInteract);
+        document.removeEventListener("touch", this._handleInteract);
         document.removeEventListener("keyup", this._handleKeyup);
     }
 
     _handleInteract = (event: {target: any}) => {
-        if (this.node && this.node.contains(event.target)) {
-            return;
-        }
         if (this.props.open) {
-            this.props.onClose();
+            this.props.onDropdownChanged(false);
         }
     };
 
     _handleKeyup = (event: KeyboardEvent) => {
         if (this.props.open && event.key === "Escape") {
             event.preventDefault();
             event.stopImmediatePropagation();
-            this.props.onClose();
+            this.props.onDropdownChanged(false);
+        }
+    };
+
+    updatePosition = () => {
+        const {openerNode} = this.props;
+        if (openerNode) {
+            //$FlowFixMe typing is weird for the return value of ref callback
+            const rect = openerNode.getBoundingClientRect();
+            this.openerBounds = {
+                top: rect.top + window.pageYOffset,
+                left: rect.left + window.pageXOffset,
+                right: rect.right,
+                width: rect.width,
+            };
         }
     };
 
+    findTranslatedPosition() {
+        if (this.openerBounds) {
+            const top = this.openerBounds.top;
+            const left = this.openerBounds.left;
+            const width = this.openerBounds.width;
+            if (this.props.alignment === "left") {
+                return {
+                    transform: `translate3d(${left}px, ${top}px, 0)`,
+                };
+            } else {
+                return {
+                    transform: `translate3d(${left +
+                        width -
+                        (this.menuWidth || 0)}px, ${top}px, 0)`,
+                };
+            }
+        }
+    }
+
+    maybeRenderMenu() {
+        const {items, light, open, style} = this.props;
+
+        this.updatePosition();
+        if (open) {
+            return (
+                <View
+                    onClick={(event) => {
+                        // Stop propagation to prevent the mouse click listener on the
+                        // document from closing the menu.
+                        event.stopPropagation();
+                        event.nativeEvent.stopImmediatePropagation();
```

Comment:
🤷‍♂️ Are both of these required? It seems like only one of them would be needed, I would've thought.